### PR TITLE
Enqueue one task per root_promise_id

### PR DESCRIPTION
**Pull request by avillega**
Enqueue a single task per root_promise_id. This is necessary to prevent extra work happening for a single promise id.

### CODE REVIEW

**Comment by avillega** [comment]
@dfarr I am still unsure how to add dst to this change, any pointers will be appreciated.

File: internal/app/subsystems/aio/store/store.go [hunk 1]
```diff
@@ -71,3 +73,13 @@ func Collect(c <-chan *bus.SQE[t_aio.Submission, t_aio.Completion], f <-chan int
 
 	return batch, true
 }
+
+func StoreErr(err error) error {
```
Comment (dfarr) [inline internal/app/subsystems/aio/store/store.go:77]: Damn, this looks really helpful!

**Comment by dfarr** [comment]
Sorry for being anal here:
```
enqueable ->
enqueueable
```

File: internal/app/subsystems/aio/store/store.go (hunk 1 above)
Comment (avillega) [reply internal/app/subsystems/aio/store/store.go:77]: I hated that the errors had no context just, `expected 3 args got 4` or something like that. This at least points to the file and line where that happens.

**Review by dfarr — APPROVED** [review]
🎉

**Pull request merged**